packages/core/src/repository/data-source/network.data-source.ts

```diff
@@ -0,0 +1,102 @@
+import { lastValueFrom, Observable } from 'rxjs';
+import { DataSource } from './data-source';
+import { Query } from '../query/query';
+import { MethodNotImplementedError, QueryNotSupportedError } from '../errors';
+import { DataSourceMapper } from './data-source-mapper';
+import { ArrayMapper, JsonDeserializerMapper, VoidMapper } from '../mapper/mapper';
+import { Type } from '../../helpers';
+import {
+    DeleteNetworkQuery,
+    GetNetworkQuery,
+    NetworkQuery,
+    PostNetworkQuery,
+    PutNetworkQuery,
+} from '../query/network-query';
+import { ApiRequestService } from './api-request.service';
+import { HttpRequestBuilder } from '../../data';
+
+export class NetworkDataSource implements DataSource<unknown> {
+    constructor(private readonly requestService: ApiRequestService) {}
+
+    public get(query: Query): Promise<unknown> {
+        if (query instanceof GetNetworkQuery) {
+            return lastValueFrom(this.getRequestWithParameters<unknown>(query).get());
+        }
+
+        throw new QueryNotSupportedError();
+    }
+
+    public put(value: unknown | undefined, query: Query): Promise<unknown> {
+        let $request: Observable<unknown>;
```

Comment:
**Nit:** isn't the convention to put the `$` at the end? So it reads like "requests"; since it's a stream, multiple values might be emitted. Although we know only one will be emitted.

See: https://angular.io/guide/rx-library#naming-conventions-for-observables

Reply:
You're right